### PR TITLE
JsonExporter: make Json export more extensible.

### PR DESCRIPTION
**Pull request by ptr1120**
Hello, 
I would like to add additional data to the existing JSON BenchmarkDotNet report, but there is currently no possibility to enrich the report. That's why I separated `ExportToLog` into report creation (`ExportToLog`) and further virtual report data generation methods (`CreateLog`).

Best regards
Peter

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/BenchmarkDotNet?pullRequest=1915) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/BenchmarkDotNet?pullRequest=1915) before we can accept your contribution.<br/><br/>:x: ptr1120 [sign now](https://cla.dotnetfoundation.org/dotnet/BenchmarkDotNet?pullRequest=1915)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/BenchmarkDotNet?pullRequest=1915) it.</sub>

**Comment by ptr1120** [comment]
Hello, did I miss anything in order to get this merged?

File: src/BenchmarkDotNet/Exporters/Json/JsonExporterBase.cs [hunk 1]
```diff
@@ -21,51 +22,69 @@ protected JsonExporterBase(bool indentJson = false, bool excludeMeasurements = f
         }
 
         public override void ExportToLog(Summary summary, ILogger logger)
+        {
+            JsonSerializer.CurrentJsonSerializerStrategy.Indent = IndentJson;
+            logger.WriteLine(JsonSerializer.SerializeObject(CreateLog(summary)));
+        }
+
+        protected virtual IReadOnlyDictionary<string, object> CreateLog(Summary summary)
```
Comment (adamsitnik) [inline src/BenchmarkDotNet/Exporters/Json/JsonExporterBase.cs:30]: nit: I am not a big fan of the `CreateLog` name. I would rather call this method `GetDataToSerialize`

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @ptr1120 !

**Pull request closed without merging**